### PR TITLE
[SPARK-22572] [Spark shell] spark-shell does not re-initialize on :replay

### PR DESCRIPTION
**Pull request by mpetruska**
## What changes were proposed in this pull request?

Ticket: [SPARK-22572](https://issues.apache.org/jira/browse/SPARK-22572)

## How was this patch tested?

Added a new test case to `org.apache.spark.repl.ReplSuite`

### CODE REVIEW

**Review by srowen — APPROVED** [review]
Aha, OK if that works that looks good.

File: repl/src/test/scala/org/apache/spark/repl/ReplSuite.scala [hunk 1]
```diff
@@ -217,4 +218,13 @@ class ReplSuite extends SparkFunSuite {
     assertDoesNotContain("error:", output)
     assertDoesNotContain("Exception", output)
   }
+
+  test(":replay should work correctly") {
+   runInterpreter("local",
+     """
+     |sc
+     |:replay
+     """.stripMargin) should not include "error: not found: value sc"
```
Comment (HyukjinKwon) [inline repl/src/test/scala/org/apache/spark/repl/ReplSuite.scala]: Could we just use assert... instead?

**Review by HyukjinKwon — APPROVED** [review]

File: repl/src/test/scala/org/apache/spark/repl/ReplSuite.scala (hunk 1 above)
Comment (mpetruska) [reply repl/src/test/scala/org/apache/spark/repl/ReplSuite.scala]: Okay, updated the test case; also removed `Matchers` as this was the only matcher used.

**Comment by HyukjinKwon** [comment]
test this please

**Comment by HyukjinKwon** [comment]
Hm, wait. Let me some minutes.

**Comment by HyukjinKwon** [comment]
LGTM

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**